src/server/search/search_family.cc

```diff
@@ -154,12 +182,27 @@ void SearchFamily::FtSearch(CmdArgList args, ConnectionContext* cntx) {
 
   size_t total_count = 0;
   for (const auto& shard_docs : docs)
-    total_count += shard_docs.size();
+    total_count += shard_docs.total_hits;
+
+  size_t response_count =
+      min(total_count - min(total_count, params->limit_offset), params->limit_total);
 
-  (*cntx)->StartArray(total_count * 2 + 1);
+  (*cntx)->StartArray(response_count * 2 + 1);
   (*cntx)->SendLong(total_count);
+
+  size_t sent = 0;
+  size_t to_skip = 0;
   for (const auto& shard_docs : docs) {
-    for (const auto& [key, doc] : shard_docs) {
+    for (const auto& [key, doc] : shard_docs.docs) {
+      // Scoring is not implemented yet, so we take just cut them in the order they were retrieved
+      if (to_skip > 0) {
+        to_skip--;
+        continue;
+      }
+
+      if (sent++ >= response_count)
+        goto END;
```

Comment:
I too wish one could do `break(2)` or something, but perhaps you could just `return` here instead?